### DISCUSSION TITLE
Nextcloud AIO stuck in upgrade

### DISCUSSION THREAD

**Discussion by Athan1** [nextcloud/all-in-one#911] [Questions] [answered]
### Steps to reproduce
1. Stop Containers 
2. Upgrade mastercontainer
3. start containers

### Expected behavior 
Nextcloud-aio should work as before after upgrading the containers 

### Actual behavior 
Getting the message: 
```
"This Nextcloud instance is currently in maintenance mode, which may take a while. This page will refresh itself when the instance is available again.

Contact your system administrator if this message persists or appeared unexpectedly."
```

When try to end the maintenance mode i get:
`Nextcloud or one of the apps require upgrade - only a limited number of commands are available`

### Host OS 
 Ubuntu 20.04.4 LTS

#### Nextcloud AIO version 
 Nextcloud AIO v1.4.2 (came from Nextcloud AIO v1.4.1)

#### Current channel 
latest

#### Other valuable info 
Normaly I would try to run ccc upgrade which won't work with AIO:
`php occ upgrade -vvv
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
2022-07-03T14:04:35+00:00 Setting log level to debug
2022-07-03T14:04:35+00:00 Repair step: Repair MySQL collation
2022-07-03T14:04:35+00:00 Repair info: Not a mysql database -> nothing to do
2022-07-03T14:04:35+00:00 Repair step: Repair SQLite autoincrement
2022-07-03T14:04:35+00:00 Repair step: Copy data from accounts table when migrating from ownCloud
2022-07-03T14:04:36+00:00 Repair step: Drop account terms table when migrating from ownCloud
2022-07-03T14:04:36+00:00 Updating database schema
2022-07-03T14:04:36+00:00 Updated database
2022-07-03T14:04:36+00:00 UnexpectedValueException: The files of the app "updatenotification" were not correctly replaced before running the update
2022-07-03T14:04:36+00:00 Update failed
2022-07-03T14:04:36+00:00 Maintenance mode is kept active
2022-07-03T14:04:36+00:00 Resetting log level`

**Comment by szaimen** [comment, marked as answer]
Hi, please create a backup, restore a backup from before the upgrade and start the containers and then restart them in order to try the upgrade again.

If the upgrade should fail again, please post the whole log of the nextcloud container here.

**Comment by szaimen** [comment]
Since no one else seems to have experienced this issue, and there is no further feedback and I cannot reproduce this, I am moving this to discussions.

**Comment by MartinM01** [comment]
Hi,
I have the same issue when migrating from nextcloud 24.0.4 docker setup to nextcloud aio 24.0.3.
File and database migration as described in https://github.com/nextcloud/all-in-one/blob/main/migration.md .
Are there any ideas how to solve this?
Is this related? https://github.com/nextcloud/all-in-one/commit/785257d798091f73e53c1841aa3336abaf142afa

**Reply by szaimen** [reply]
> have the same issue when migrating from nextcloud 24.0.4 docker setup to nextcloud aio 24.0.3.

This is not supported and the reason why it fails for you. It needs to match the exact version.

**Reply by MartinM01** [reply]
hm ok so this is because of the donwgrade?

**Reply by szaimen** [reply]
Yes

**Reply by MartinM01** [reply]
is 24.0.4 currently in beta?

**Reply by szaimen** [reply]
yes

**Reply by MartinM01** [reply]
ok great I try this :), thx

**Comment by MartinM01** [comment]
nc: getaddrinfo: Name does not resolve
Waiting for database to start...
              now              
-------------------------------
 2022-08-19 07:38:22.897746+00
(1 row)

Configuring Redis as session handler
Applying one-click-instance settings...
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value one-click-instance set to boolean true
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value one-click-instance.user-limit set to integer 100
Adjusting log files...
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value logfile set to string /var/www/html/data/nextcloud.log
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value logfile for app admin_audit set to /var/www/html/data/audit.log
Applying network settings...
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value trusted_domains => 1 set to string xxxx
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value overwrite.cli.url set to string https://xxxx/
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value htaccess.RewriteBase set to string /
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
.htaccess has been updated
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value files_external_allow_create_new_local set to boolean true
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
nextcloud-aio 0.2.0 enabled
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value trusted_proxies => 0 set to string 127.0.0.1
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value trusted_proxies => 1 set to string ::1
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value base_endpoint for app notify_push set to https://xxxx/push
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value wopi_url for app richdocuments set to https://xxxx/
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
System config value allow_local_remote_servers set to boolean true
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value stun_servers for app spreed set to ["nextcloud2.meshub.de:3480"]
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value turn_servers for app spreed set to [{"server":"xxxx","secret":"31d9596ebb8a34c4511cfc7c72d6228d8910f0c9573972dd","protocols":"udp,tcp"}]
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value signaling_servers for app spreed set to {"servers":[{"server":"xxxx/standalone-signaling/","verify":true}],"secret":"fde175238c83faece25a1c4f9210869996ea24732c624d38"}
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value av_mode for app files_antivirus set to daemon
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value av_port for app files_antivirus set to 3310
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value av_host for app files_antivirus set to nextcloud-aio-clamav
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value av_stream_max_length for app files_antivirus set to 104857600
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value av_max_file_size for app files_antivirus set to -1
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Config value av_infected_action for app files_antivirus set to only_log
Activating collabora config
[19-Aug-2022 07:38:31] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[19-Aug-2022 07:38:31] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[19-Aug-2022 07:38:31] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[19-Aug-2022 07:38:31] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[19-Aug-2022 07:38:31] NOTICE: fpm is running, pid 264
[19-Aug-2022 07:38:31] NOTICE: ready to handle connections
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade

                                                                   
  There are no commands defined in the "richdocuments" namespace.  
  
  ------
  
docker exec --user www-data nextcloud-aio-nextcloud php occ upgrade
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Setting log level to debug
Updating database schema
Updated database
UnexpectedValueException: The files of the app "updatenotification" were not correctly replaced before running the update
Update failed
Maintenance mode is kept active
Resetting log level

**Comment by MartinM01** [comment]
Thanks Szaimen, with the beta version the migration was working. I will wait till release to bring this live.

Some points to mention:
1.
I´d try to run a second instance on my server to validate nextcloud aio.
I was thinking if I use a proxy and change the talk port I could run 2 nextcloud instances. 1x live and 1x validation.
But Nextcloud Talk seems to use aditional not configurable ports:
```
---------------------------------------------------
  Starting Meetecho Janus (WebRTC Server) v0.11.2
---------------------------------------------------

Checking command line arguments...
Debug/log level is 4
Debug/log timestamps are disabled
Debug/log colors are disabled
Adding 'vmnet' to the ICE ignore list...
Using 172.17.0.4 as local IP...
Token based authentication disabled
Initializing recorder code
Initializing ICE stuff (Full mode, ICE-TCP candidates disabled, full-trickle, IPv6 support disabled)
STUN server to use: 127.0.0.1:3480
  >> 127.0.0.1:3480 (IPv4)
Testing STUN server: message is of 20 bytes
  >> Our public address is 127.0.0.1
TURN server to use: 127.0.0.1:3480 (udp)
TURN REST API backend: (disabled)
Crypto: OpenSSL >= 1.1.0
No cert/key specified, autogenerating some...
Fingerprint of our certificate: C6:E3:32:9F:FA:98:41:9B:AC:7A:76:2C:70:10:32:B9:E1:6A:5A:4E:00:87:A3:03:F7:54:83:8B:75:2B:A6:9B
Event handlers support disabled
Plugins folder: /usr/lib/x86_64-linux-gnu/janus/plugins
Joining Janus requests handler thread
Sessions watchdog started
Loading plugin 'libjanus_echotest.so'...
JANUS EchoTest plugin initialized!
Loading plugin 'libjanus_recordplay.so'...
JANUS Record&Play plugin initialized!
Loading plugin 'libjanus_textroom.so'...
JANUS TextRoom plugin initialized!
Loading plugin 'libjanus_nosip.so'...
JANUS NoSIP plugin initialized!
Loading plugin 'libjanus_videocall.so'...
JANUS VideoCall plugin initialized!
Loading plugin 'libjanus_audiobridge.so'...
JANUS AudioBridge plugin initialized!
Loading plugin 'libjanus_duktape.so'...
[[36mechotest.js[0m]  Loading script...
[[36mechotest.js[0m]  Modules folder: /usr/share/janus/duktape
[[36mechotest.js[0m]  Loading module: janus-sdp
[[36mechotest.js[0m]  Module loaded
[[36mechotest.js[0m]  Script loaded
[[36mechotest.js[0m]  Initializing...
[[36mechotest.js[0m]  Initialized
Janus JavaScript plugin (Duktape) initialized!
Loading plugin 'libjanus_voicemail.so'...
JANUS VoiceMail plugin initialized!
Loading plugin 'libjanus_sip.so'...
JANUS SIP plugin initialized!
Loading plugin 'libjanus_streaming.so'...
[ERR] [plugins/janus_streaming.c:janus_streaming_create_fd:5574] [rtp-sample] Bind failed for audio (port 5002)... 98 (Address already in use)
[ERR] [plugins/janus_streaming.c:janus_streaming_create_rtp_source:5783] Can't bind to port 5002 for audio...
[ERR] [plugins/janus_streaming.c:janus_streaming_init:1956] Error creating 'rtp' mountpoint 'rtp-sample'...
[ERR] [plugins/janus_streaming.c:janus_streaming_init:2005] Can't add 'live' mountpoint, no such file '/usr/share/janus/streams/radio.alaw'...
[ERR] [plugins/janus_streaming.c:janus_streaming_init:2072] Can't add 'ondemand' mountpoint, no such file '/usr/share/janus/streams/music.mulaw'...
JANUS Streaming plugin initialized!
Loading plugin 'libjanus_videoroom.so'...
JANUS VideoRoom plugin initialized!
Loading plugin 'libjanus_lua.so'...
[0m[[34mechotest.lua[0m][0m Loading...
[0m[[34mechotest.lua[0m][0m Loaded
[0m[[34mechotest.lua[0m][0m Initializing...
[0m[[34mechotest.lua[0m][0m Initialized
Janus Lua plugin initialized!
Transport plugins folder: /usr/lib/x86_64-linux-gnu/janus/transports
Loading transport plugin 'libjanus_mqtt.so'...
[ERR] [config.c:janus_config_parse:205] Error parsing config file at line 30: syntax error
[WARN] Couldn't find .jcfg configuration file (janus.transport.mqtt), trying .cfg
[ERR] [config.c:janus_config_parse:191]   -- Error reading configuration file 'janus.transport.mqtt.cfg'... error 2 (No such file or directory)
MQTT SSL support disabled
[WARN] MQTT support disabled for both Janus and Admin API, giving up
JANUS MQTT transport plugin destroyed!
[WARN] The 'janus.transport.mqtt' plugin could not be initialized
Loading transport plugin 'libjanus_nanomsg.so'...
[ERR] [transports/janus_nanomsg.c:janus_nanomsg_init:224] Error binding Janus API Nanomsg socket to address 'ipc:///tmp/janus.ipc': 98 (Address already in use)
[WARN] No Nanomsg server started, giving up...
[WARN] The 'janus.transport.nanomsg' plugin could not be initialized
Loading transport plugin 'libjanus_pfunix.so'...
[WARN] No Unix Sockets server started, giving up...
[WARN] The 'janus.transport.pfunix' plugin could not be initialized
Loading transport plugin 'libjanus_http.so'...
HTTP transport timer started
[FATAL] [transports/janus_http.c:janus_http_init:795] Couldn't start webserver on port 8088...
[WARN] No HTTP/HTTPS server started, giving up...
[WARN] The 'janus.transport.http' plugin could not be initialized
Loading transport plugin 'libjanus_rabbitmq.so'...
RabbitMQ SSL support disabled
[WARN] RabbitMQ support disabled for both Janus and Admin API, giving up
[WARN] The 'janus.transport.rabbitmq' plugin could not be initialized
Loading transport plugin 'libjanus_websockets.so'...
libwebsockets logging: 0
[FATAL] [transports/janus_websockets.c:janus_websockets_init:679] Error creating vhost for WebSockets server...
[WARN] No WebSockets server started, giving up...
[WARN] The 'janus.transport.websockets' plugin could not be initialized
[FATAL] [janus.c:main:5447] No Janus API transport is available... enable at least one and restart Janus
Error launching Janus (error code 1), check the logs for more details
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
hub.go:2307: Could not upgrade request from 172.17.0.1: websocket: the client is not using the websocket protocol: 'upgrade' token not found in 'Connection' header
```
2.
Some error, but this could be because it is beta:
```
Waiting for database to start...
Waiting for database to start...
              now              
-------------------------------
 2022-08-19 13:39:43.840383+02
(1 row)

Configuring Redis as session handler
Applying one-click-instance settings...
System config value one-click-instance set to boolean true
System config value one-click-instance.user-limit set to integer 100
Adjusting log files...
System config value logfile set to string /var/www/html/data/nextcloud.log
Config value logfile for app admin_audit set to /var/www/html/data/audit.log
Applying network settings...
System config value trusted_domains => 1 set to string xxxx
System config value overwrite.cli.url set to string https://xxxx/
System config value htaccess.RewriteBase set to string /
.htaccess has been updated
System config value files_external_allow_create_new_local set to boolean true
nextcloud-aio 0.2.0 enabled
System config value trusted_proxies => 0 set to string 127.0.0.1
System config value trusted_proxies => 1 set to string ::1
Config value base_endpoint for app notify_push set to https://xxxx/push
Config value wopi_url for app richdocuments set to https://xxxx/
System config value allow_local_remote_servers set to boolean true
Config value stun_servers for app spreed set to ["xxxx:3480"]
Config value turn_servers for app spreed set to [{"server":"xxxx:3480","secret":"d454586f780d1c3a3e4855b50c5cec1820182b5785e8f2ed","protocols":"udp,tcp"}]
Config value signaling_servers for app spreed set to {"servers":[{"server":"https://xxxx/standalone-signaling/","verify":true}],"secret":"0058a0f1e05f65d156d6fe24b0292c25fb48d64c5d5313d2"}
Config value av_mode for app files_antivirus set to daemon
Config value av_port for app files_antivirus set to 3310
Config value av_host for app files_antivirus set to nextcloud-aio-clamav
Config value av_stream_max_length for app files_antivirus set to 104857600
Config value av_max_file_size for app files_antivirus set to -1
Config value av_infected_action for app files_antivirus set to only_log
Activating collabora config
[19-Aug-2022 13:39:55] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[19-Aug-2022 13:39:55] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[19-Aug-2022 13:39:55] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[19-Aug-2022 13:39:55] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
[19-Aug-2022 13:39:55] NOTICE: fpm is running, pid 268
[19-Aug-2022 13:39:55] NOTICE: ready to handle connections
Failed to activate any config changes
Server error: `GET https://xxxx/hosting/discovery` resulted in a `502 Bad Gateway` response:
<html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>ngin (truncated...)

#0 /var/www/html/3rdparty/guzzlehttp/guzzle/src/Middleware.php(69): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL)
#1 /var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php(204): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response))
#2 /var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php(153): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL)
#3 /var/www/html/3rdparty/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
#4 /var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Promise\TaskQueue->run(true)
#5 /var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn()
#6 /var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
#7 /var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
#8 /var/www/html/3rdparty/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending()
#9 /var/www/html/3rdparty/guzzlehttp/guzzle/src/Client.php(187): GuzzleHttp\Promise\Promise->wait()
#10 /var/www/html/lib/private/Http/Client/Client.php(218): GuzzleHttp\Client->request('get', 'https://nextclo...', Array)
#11 /var/www/html/custom_apps/richdocuments/lib/WOPI/DiscoveryManager.php(84): OC\Http\Client\Client->get('https://nextclo...', Array)
#12 /var/www/html/custom_apps/richdocuments/lib/WOPI/DiscoveryManager.php(56): OCA\Richdocuments\WOPI\DiscoveryManager->fetchFromRemote()
#13 /var/www/html/custom_apps/richdocuments/lib/WOPI/Parser.php(41): OCA\Richdocuments\WOPI\DiscoveryManager->get()
#14 /var/www/html/custom_apps/richdocuments/lib/Command/ActivateConfig.php(70): OCA\Richdocuments\WOPI\Parser->getUrlSrc('Capabilities')
#15 /var/www/html/3rdparty/symfony/console/Command/Command.php(255): OCA\RichDocuments\Command\ActivateConfig->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /var/www/html/3rdparty/symfony/console/Application.php(1009): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /var/www/html/3rdparty/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand(Object(OCA\RichDocuments\Command\ActivateConfig), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/html/3rdparty/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /var/www/html/lib/private/Console/Application.php(211): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 /var/www/html/console.php(100): OC\Console\Application->run()
#21 /var/www/html/occ(11): require_once('/var/www/html/c...')
#22 {main}
```

**Reply by szaimen** [reply]
Did you read https://github.com/nextcloud/all-in-one/blob/main/multiple-instances.md?

**Reply by MartinM01** [reply]
Yes but I don't use a vm for every docker "app". That's why all apps share the localhost interface.

It looks like nextcloud aio uses some fixed ports there. E.g. the 5002 which can't be opened again in my log.

**Reply by szaimen** [reply]
If you want to run multiple instances of AIO on the server, you need to follow this documentation.

I don't see port 5002 anywhere...
https://github.com/nextcloud/all-in-one/search?q=5002

**Reply by MartinM01** [reply]
[ERR] [plugins/janus_streaming.c:janus_streaming_create_fd:5574] [rtp-sample] Bind failed for audio (port 5002)... 98 (Address already in use)

**Reply by MartinM01** [reply]
udp6       0      0 :::5002                 :::*                                3978/janus
udp6       0      0 :::5004                 :::*                                3978/janus

**Reply by MartinM01** [reply]
raw        0      0 0.0.0.0:132             0.0.0.0:*               7           3978/janus
raw6       0      0 :::132                  :::*                    7           3978/janus
